### PR TITLE
runtimevar: implement health.Checker

### LINKED ISSUE #1443: runtimevar: implement the health.Checker interface for Variable
Returning healthy only when there's a "good" value for the variable.

### PR DESCRIPTION
**Pull request by vangent**
Fixes #1443.

### CODE REVIEW

File: runtimevar/runtimevar.go [hunk 1]
```diff
@@ -35,6 +35,8 @@
 // deploy it to multiple Cloud providers. You may find
 // http://github.com/google/wire useful for managing your initialization code.
 //
+// Variable implements health.Checker; it reports as healthy when Latest will
+// return a value.
```
Comment (zombiezen) [inline runtimevar/runtimevar.go]:
```suggestion
// return a value without blocking.
```

File: runtimevar/runtimevar_test.go [hunk 2]
```diff
@@ -202,6 +202,10 @@ func TestVariable_Latest(t *testing.T) {
 	if ctx2.Err() == nil {
 		t.Error("Latest with non-nil ctx but no value yet should block")
 	}
+	// And we're not healthy.
+	if v.CheckHealth() == nil {
```
Comment (zombiezen) [inline runtimevar/runtimevar_test.go]: Add a health check right after `Variable` creation, too.

File: runtimevar/runtimevar.go [hunk 3]
```diff
@@ -278,6 +280,23 @@ func (c *Variable) Latest(ctx context.Context) (Snapshot, error) {
...
+	case <-c.haveGood:
+		haveGood = true
+	default:
+	}
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if haveGood && c.lastErr != ErrClosed {
+		return nil
+	}
+	return c.lastErr
```
Comment (zombiezen) [inline runtimevar/runtimevar.go]: Is `c.lastErr` guaranteed to be non-nil? What if the RPC hasn't completed yet?

File: runtimevar/runtimevar_test.go (hunk 2 above)
Comment (vangent) [reply runtimevar/runtimevar_test.go]: Done.

File: runtimevar/runtimevar.go (hunk 3 above)
Comment (vangent) [reply runtimevar/runtimevar.go]: `c.lastErr` is initialized to `errors.New("no value yet")` in the constructor, so it's always non-nil unless there's a good value.

**Pull request merged**